src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,845 @@
+                                         mEndpointId));
+    }
+
+    if (SupportsOptAttr(OptionalAttributes::kPanMin))
+    {
+        VerifyOrReturnError(HasFeature(Feature::kMechanicalPan), CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl,
+                                         "CameraAVSettingsUserLevelManagement[ep=%d]: Feature configuration error. If PanMin is "
+                                         "enabled, then Mechanical Pan feature is required",
+                                         mEndpointId));
+    }
+
+    if (SupportsOptAttr(OptionalAttributes::kPanMax))
+    {
+        VerifyOrReturnError(HasFeature(Feature::kMechanicalPan), CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl,
+                                         "CameraAVSettingsUserLevelManagement[ep=%d]: Feature configuration error. If PanMax is "
+                                         "enabled, then Mechanical Pan feature is required",
+                                         mEndpointId));
+    }
+
+    LoadPersistentAttributes();
+
+    // Set default MPTZ
+    mMptzPosition.pan  = defaultPan;
+    mMptzPosition.tilt = defaultTilt;
+    mMptzPosition.zoom = defaultZoom;
+
+    VerifyOrReturnError(AttributeAccessInterfaceRegistry::Instance().Register(this), CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(CommandHandlerInterfaceRegistry::Instance().RegisterCommandHandler(this));
+    return CHIP_NO_ERROR;
+}
+
+bool CameraAvSettingsUserLevelMgmtServer::HasFeature(Feature feature) const
+{
+    return mFeature.Has(feature);
+}
+
+bool CameraAvSettingsUserLevelMgmtServer::SupportsOptAttr(OptionalAttributes aOptionalAttrs) const
+{
+    return mOptionalAttrs.Has(aOptionalAttrs);
+}
+
+// Mutators that may be invoked by a delegate in responding to command callbacks, or due to local on device changes
+//
+void CameraAvSettingsUserLevelMgmtServer::setPan(Optional<int16_t> pan)
+{
+    if (pan.HasValue())
+    {
+        mMptzPosition.pan = pan;
+    }
+}
+
+void CameraAvSettingsUserLevelMgmtServer::setTilt(Optional<int16_t> tilt)
+{
+    if (tilt.HasValue())
+    {
+        mMptzPosition.tilt = tilt;
+    }
+}
+
+void CameraAvSettingsUserLevelMgmtServer::setZoom(Optional<int8_t> zoom)
+{
+    if (zoom.HasValue())
+    {
+        mMptzPosition.zoom = zoom;
+    }
+}
+
+// Helper function for setting the next preset ID to use
+//
+void CameraAvSettingsUserLevelMgmtServer::UpdatePresetID()
+{
+
+    // Has the next possible incremented ID been used by a user set Preset?
+    //
+    uint8_t increment = 1;
+    do
+    {
+        auto it = std::find_if(mMptzPresetHelper.begin(), mMptzPresetHelper.end(), [=](const MPTZPresetHelper & mptzph) {
+            return mptzph.GetPresetID() == currentPresetID + increment;
+        });
+        if (it == mMptzPresetHelper.end())
+        {
+            currentPresetID = static_cast<uint8_t>(currentPresetID + increment);
```

Comment:
Why is `currentPresetID` not named like the other members?  `mCurrentPresetID` or so.

Reply:
Renamed.